src/scripts/create-parameters.py

```diff
@@ -79,9 +52,45 @@ def check_mapping(m):
 def convert_mapping(m):
   return [m[1], json.loads(m[2])]
 
-mappings = filter(check_mapping, mappings)
-mappings = map(convert_mapping, mappings)
-mappings = dict(mappings)
+def write_mappings(mappings, output_path):
+  with open(output_path, 'w') as fp:
+    fp.write(json.dumps(mappings))
+
+def create_parameters(output_path, head, base, mapping):
+  checkout(base)  # Checkout base revision to make sure it is available for comparison
+  checkout(head)  # return to head commit
+  base = merge_base(base, head)
+
+  if head == base:
+    try:
+      # If building on the same branch as BASE_REVISION, we will get the
+      # current commit as merge base. In that case try to go back to the
+      # first parent, i.e. the last state of this branch before the
+      # merge, and use that as the base.
+      base = parent_commit()
+    except:
+      # This can fail if this is the first commit of the repo, so that
+      # HEAD~1 actually doesn't resolve. In this case we can compare
+      # against this magic SHA below, which is the empty tree. The diff
+      # to that is just the first commit as patch.
+      base = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
+
+  print('Comparing {}...{}'.format(base, head))
+  changes = changed_files(base, head)
+
+  mappings = [
+    m.split() for m in
+    os.environ.get('MAPPING').splitlines()
+  ]
+  mappings = filter(partial(check_mapping, changes), mappings)
+  mappings = map(convert_mapping, mappings)
+  mappings = dict(mappings)
+
+  write_mappings(mappings, output_path)
```

Comment:
\~~~ It might be easier to unit test in the future if it returned the mappings as a string rather than writing to a file.